crates/turbo-tasks-memory/src/gc.rs

```diff
@@ -88,15 +95,24 @@ pub struct GcQueue {
     /// Tasks from old generations. The oldest generation will be garbage
     /// collected next.
     generations: Mutex<VecDeque<OldGeneration>>,
+    /// Tasks that have become inactive. Processing them should ensure them for
+    /// GC, if they are not already ensured and put all child tasks into the
+    /// activation_queue
+    deactivation_queue: ConcurrentQueue<TaskId>,
+    /// Tasks that are active and not enqueued in the deactivation queue.
+    // TODO Could be a bit field with locks, an array of atomics or an AMQF.
+    active_tasks: DashSet<TaskId>,
 }
 
 impl GcQueue {
     pub fn new() -> Self {
         Self {
-            generation: AtomicU32::new(0),
+            generation: AtomicU32::new(1),
```

Comment:
Why start at 1 here?

Reply:
we use generation 0 as marker for "not in any generation", e. g. after as task was GC'ed it will have generation set to 0.

I can change it to Option<NonZeroU32> to make it more clear.